stdio/file.c

```diff
@@ -146,9 +146,22 @@ static ssize_t safe_write(int fd, const void *buf, size_t size)
 
 static int safe_open(const char *path, int oflag, mode_t mode)
 {
+	struct stat s;
 	int err;
-	while ((err = open(path, oflag, mode)) < 0 && errno == EINTR)
+
+	if (oflag & (O_WRONLY | O_RDWR | O_APPEND)) {
```

Comment:
`O_APPEND` should not be here, POSIX requires to always use one of the basic open flags (`O_APPEND` is an additional one)

Reply:
Changed as requested. Please note that in that case we won't properly handle fopen() on an existing directory with "a" mode. The "a" mode gets translated to O_APPEND | O_CREAT (so doesn't pass our write access check) and we will successfully open directory in append mode.
I did check this case on Linux - it does set errno to EISDIR and return -1 in that case.

Reply:
Actually, shouldn't append mode ("a") [here](https://github.com/phoenix-rtos/libphoenix/blob/bd09e6dc360d76ef1b9facda9ecefa335aa5e58a/stdio/file.c#L83) be translated to O_APPEND | O_WRONLY | O_CREAT?
That would solve the issue.

Reply:
Added append mode flags in another [PR](https://github.com/phoenix-rtos/libphoenix/pull/146)